priv/templates/phx.gen.schema/schema.ex

```diff
@@ -2,8 +2,9 @@ defmodule <%= inspect schema.module %> do
   use Ecto.Schema
   import Ecto.Changeset
 <%= if schema.prefix do %>
-  @schema_prefix :<%= schema.prefix %><% end %><%= if schema.binary_id do %>
-  @primary_key {:id, :binary_id, autogenerate: true}
+  @schema_prefix :<%= schema.prefix %><% end %><%= if schema.opts[:primary_key] do %>
+  @derive {Phoenix.Param, key: :<%= schema.opts[:primary_key] %>}<% end %><%= if schema.binary_id do %>
+  @primary_key {:<%= schema.opts[:primary_key] || :id %>, :binary_id, autogenerate: true}
```

Comment:
I think we also need an else case if we have a custom primary key name, but not using binary ids?

Reply:
You are right.  I'll update and test in a little bit.